File: 2616-maximal-score-after-applying-k-operations/maximal-score-after-applying-k-operations.java
class Solution {
    public long maxKelements(int[] nums, int k) {
        long ans = 0;
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());

        for(int i=0; i<nums.length; i++){
            pq.add(nums[i]);
        }

        for(int i=0; i<k; i++){
            int tmp = pq.peek();
            // System.out.println(tmp + " " + Math.ceil(tmp/3.0));
            pq.remove(tmp);
            pq.add((int) Math.ceil(tmp/3.0));
            ans += tmp;
        }

        return ans;
    }
}

File: 2616-maximal-score-after-applying-k-operations/maximal-score-after-applying-k-operations.cpp
class Solution {
public:
    long long maxKelements(vector<int>& nums, int k) {
        long long ans = 0;

        multiset<int> st;

        for(int i=0; i<nums.size(); i++){
            st.insert(nums[i]);
        }

        for(int i=0; i<k; i++){
            auto last = prev(st.end());
            // cout << *last << endl;
            int tmp = *last;
            st.erase(last);
            ans += tmp;
            st.insert(ceil(tmp/3.0));
        }

        return ans;
    }
};